087-external-certificate-manager.md

```diff
@@ -0,0 +1,241 @@
+2. Create an `Issuer` or `ClusterIssuer` custom resource.
+3. Create a `Secret` containing the CA public cert for Strimzi to trust.
+   Users can optionally use [trust-manager](https://cert-manager.io/docs/trust/trust-manager/) to create this Secret, but they are responsible for installing trust-manager and creating the `Bundle` CR with a single `sources` entry.
+4. Create a `Kafka` resource with `clusterCa.type` and/or `clientsCa.type` set to `cert-manager.io`, and `certManagerIssuerRef` and `publicCert` configured.
+
+### Internal handling of certificates
+
+Strimzi uses two generation annotations to track how Cluster CA certificates are trusted and used:
+* `strimzi.io/cluster-ca-key-generation` on Kafka pods to indicate the CA generation trusted by that pod
+* `strimzi.io/cluster-ca-cert-generation` on Secrets containing certs to indicate the CA generation that signed those certs
+
+The `strimzi.io/cluster-ca-cert-generation` annotation is also used on Kafka pods to indicate the CA generation that signed the certs it is currently presenting (i.e. the CA generation currently in use).
+
+When using cert-manager for issuing certificates Strimzi will continue to use these generations in the same way.
+Rather than directly mounting cert-manager Secrets and the user-provided CA cert Secret, Strimzi will copy the certificates into its own existing Secrets.
+
+#### Handling Cluster CA trust rollout
+
+When a new Kafka cluster is created Strimzi will copy the CA public cert from the Secret identified in `clusterCa.publicCert` and `clientsCa.publicCert` to the `<CLUSTER_NAME>-cluster-ca-cert` and `<CLUSTER_NAME>-clients-ca-cert` Secrets respectively.
+Strimzi will add three annotations to the `<CLUSTER_NAME>-cluster-ca-cert` Secret:
+* `strimzi.io/ca-cert-hash` with the value being a hash of the certificate.
+* `strimzi.io/cluster-ca-key-generation` initially set to 0.
+* `strimzi.io/cluster-ca-cert-generation` initially set to 0.
+
+During a reconciliation Strimzi will check the hash of the certificate stored in the user's CA public cert Secret to see if an update is needed.
+If the certificate has changed Strimzi will rename the existing cert in the `<CLUSTER_NAME>-cluster-ca-cert` Secret to `ca-YYYY-MM-DDTHH-MM-SSZ.crt` and copy over the new certificate.
+It will also update the `strimzi.io/ca-cert-hash` and increment the `strimzi.io/cluster-ca-key-generation` and `strimzi.io/cluster-ca-cert-generation` annotations.
+
+#### Issuing end-entity certificates
+
+If the user has enabled this feature, when Strimzi needs to issue a certificate, instead of using the existing internal mechanism it will create a `Certificate` custom resource.
+Strimzi will specify the required CN/SANS in the `Certificate` resource for the end-entity certificate.
+Strimzi will specify the Secret for the certificate to be stored in as the existing name for the certificate Secret suffixed with `-cm`, for example `<CLUSTER_NAME>-cluster-operator-certs-cm`.
```

Comment:
I had hard time to understand what this sentence means sorry. Maybe just a language barrier on my side :-(
Can you elaborate here or make it clearer to me please?

Reply:
I've rewritten this sentence, see if it makes more sense now